agent/status.go

```diff
@@ -105,6 +108,35 @@ func (a *Agent) statusHandler() http.HandlerFunc {
 	}
 }
 
+type TriggerEndTaskResp struct {
+	apimodels.TaskEndDetail `json:",omitempty"`
+	ShouldContinue          bool `json:"should_continue"`
+}
+
+func (a *Agent) endTaskHandler(w http.ResponseWriter, r *http.Request) {
+	defer func() {
+		_ = grip.GetSender().Close()
+	}()
+
+	payload, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		_, _ = w.Write([]byte(err.Error()))
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	resp := TriggerEndTaskResp{}
+	if err := json.Unmarshal(payload, &resp); err != nil {
+		grip.Error(message.WrapError(err, message.Fields{
```

Comment:
Should these be `w.Write` instead so that the caller sees it?

Reply:
I wasn't actually aware this would reach the caller. Fixed but not sure if plain text is the best way to do it.